dask_sql/physical/rel/logical/limit.py

```diff
@@ -30,28 +30,33 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
         limit = RexConverter.convert(rel, rel.limit().getFetch(), df, context=context)
         offset = RexConverter.convert(rel, rel.limit().getSkip(), df, context=context)
 
-        # Apply Offset to DataFrame
-        df = self._apply_offset(df, offset) if offset else df
-
-        # Limit DataFrame
-        df = df.head(limit, npartitions=-1, compute=False) if limit else df
+        # apply offset to limit if specified
+        if limit and offset:
+            limit += offset
 
+        # apply limit and/or offset to DataFrame
+        df = self._apply_limit(df, limit, offset)
         cc = self.fix_column_to_row_type(cc, rel.getRowType())
+
         # No column type has changed, so no need to cast again
         return DataContainer(df, cc)
 
-    def _apply_offset(self, df: dd.DataFrame, offset: int) -> dd.DataFrame:
+    def _apply_limit(self, df: dd.DataFrame, limit: int, offset: int) -> dd.DataFrame:
         """
-        Limit the dataframe to the window [offset, end].
+        Limit the dataframe to the window [offset, limit].
 
         Unfortunately, Dask does not currently support row selection through `iloc`, so this must be done using a custom partition function.
         However, it is sometimes possible to compute this window using `head` when an `offset` is not specified.
         """
+        # if no offset is specified we can use `head` to compute the window
+        if not offset:
+            return df.head(limit, npartitions=-1, compute=False)
```

Comment:
@ayushdg not sure if you'd prefer we check the first partition first to see if we can avoid the `npartitions=-1` here?

Reply:
I think it's fine to keep it as is and revisit if we encounter perf issues.